CRM/Core/Task.php

```diff
@@ -0,0 +1,184 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | CiviCRM version 4.7                                                |
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC (c) 2004-2017                                |
+ +--------------------------------------------------------------------+
+ | This file is a part of CiviCRM.                                    |
+ |                                                                    |
+ | CiviCRM is free software; you can copy, modify, and distribute it  |
+ | under the terms of the GNU Affero General Public License           |
+ | Version 3, 19 November 2007 and the CiviCRM Licensing Exception.   |
+ |                                                                    |
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC (c) 2004-2017
+ */
+
+/**
+ * Class to represent the actions that can be performed on a group of contacts used by the search forms.
+ */
+abstract class CRM_Core_Task {
+  const
+    // General (Implemented by more than one entity)
+    GROUP_REMOVE = 1,
+    GROUP_ADD = 2,
+    PDF_LETTER = 3,
+    TASK_DELETE = 4,
+    TASK_PRINT = 5,
+    BATCH_UPDATE = 6,
+    TASK_SMS = 7,
+    TASK_EXPORT = 8,
+    TASK_EMAIL = 9,
+    TAG_ADD = 10,
+    TAG_REMOVE = 11,
+    // Contact tasks
+    SAVE_SEARCH = 12,
+    SAVE_SEARCH_UPDATE = 13,
+    CREATE_MAILING = 14,
+    DELETE_PERMANENTLY = 15,
+    LABEL_CONTACTS = 16;
+
+
+  /**
+   * The task array
+   *
+   * @var array
+   */
+  public static $_tasks = NULL;
+
+  static $objectType = NULL;
+
+  public static function tasks() {
```

Comment:
This function could use a docblock.

Reply:
@colemanw Agreed.  I've gone through and added more comments/docblocks